src/socket.rs

```diff
@@ -11,53 +11,72 @@ use critical_section::Mutex;
 use no_std_net::SocketAddr;
 use num_enum::{IntoPrimitive, TryFromPrimitive};
 
-const WAKER_INIT: Option<(Waker, i32)> = None;
-static SOCKET_WAKERS: Mutex<
-    RefCell<[Option<(Waker, i32)>; nrfxlib_sys::NRF_MODEM_MAX_SOCKET_COUNT as usize]>,
-> = Mutex::new(RefCell::new(
-    [WAKER_INIT; nrfxlib_sys::NRF_MODEM_MAX_SOCKET_COUNT as usize],
-));
+// use 16 slots for wakers instead of 8, which is the max number of sockets allowed, so they
+// are not overwritten when split into their rx/tx counterparts and run in separate tasks.
+const WAKER_SLOTS: usize = (nrfxlib_sys::NRF_MODEM_MAX_SOCKET_COUNT * 2) as usize;
+const WAKER_INIT: Option<(Waker, i32, i32)> = None;
+
+static SOCKET_WAKERS: Mutex<RefCell<[Option<(Waker, i32, i32)>; WAKER_SLOTS]>> =
```

Comment:
We don't have to store a number here. We can store the enum itself directly.
That way, Rust will pick the best size (likely 1 byte) which saves us some RAM.